openapi-comparator-tests/src/test/scala/sttp/apispec/openapi/validation/OpenAPIComparatorTest.scala

```diff
@@ -0,0 +1,466 @@
+package sttp.apispec.openapi.validation
+
+import io.circe
+import org.scalatest.funsuite.AnyFunSuite
+import sttp.apispec.SchemaType
+import sttp.apispec.openapi.{OpenAPI, ResponsesCodeKey}
+import sttp.apispec.openapi.circe.openAPIDecoder
+import sttp.apispec.test.ResourcePlatform
+import sttp.apispec.validation.TypeMismatch
+
+class OpenAPIComparatorTest extends AnyFunSuite with ResourcePlatform {
+  override val basedir = "openapi-comparator-tests"
+
+  def readOpenAPI(path: String): Either[circe.Error, OpenAPI] = readJson(path).flatMap(_.as[OpenAPI]): @unchecked
+  private def compare(clientOpenapi: OpenAPI, serverOpenapi: OpenAPI): List[OpenAPICompatibilityIssue] =
+    new OpenAPIComparator(clientOpenapi, serverOpenapi)
```

Comment:
minor UX: maybe `OpenAPIComparator.apply(client, server)` would be nicer? The `OpenAPIComparator` class could then have a private constructor with all the implementation details as now